### PR TITLE
Plugin samples examples

### PR DESCRIPTION
**Pull request by MartyLake**

### CODE REVIEW

File: examples/PlugInSamples/NoiseGate/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+  # BINARYDATA_NAMESPACE
+  # PREPROCESSOR_DEFINITIONS
+  PROJECT_ID "kmYCMo"
+)
+
+jucer_audio_plugin_settings(
+  BUILD_VST ON
+  BUILD_AUDIOUNIT ON
+  PLUGIN_NAME "NoiseGate"
+  PLUGIN_DESCRIPTION "NoiseGate"
```
Comment (McMartin) [inline examples/PlugInSamples/NoiseGate/CMakeLists.txt]: @MartyLake: in https://github.com/julianstorer/JUCE/blob/master/examples/PlugInSamples/NoiseGate/NoiseGate.jucer#L6, the plugin name and description are `Noise Gate`.

File: examples/PlugInSamples/NoiseGate/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+  PLUGIN_MIDI_OUTPUT OFF
+  MIDI_EFFECT_PLUGIN OFF
+  KEY_FOCUS ON
+  PLUGIN_AU_EXPORT_PREFIX "NoiseGateAU"
+  # PLUGIN_AU_MAIN_TYPE
+  # VST_CATEGORY
+)
+
+jucer_project_files("NoiseGate/Source"
+    "${NoiseGate_jucer_DIR}/Source/NoiseGate.cpp"
```
Comment (McMartin) [inline examples/PlugInSamples/NoiseGate/CMakeLists.txt]: @MartyLake: please indent with 2 spaces, thanks!

File: examples/PlugInSamples/Surround/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+  MIDI_EFFECT_PLUGIN OFF
+  KEY_FOCUS ON
+  PLUGIN_AU_EXPORT_PREFIX "SurroundAU"
+  # PLUGIN_AU_MAIN_TYPE
+  # VST_CATEGORY
+)
+
+jucer_project_files("Surround/Source"
+    "${Surround_jucer_DIR}/Source/SurroundProcessor.cpp"
+    "${Surround_jucer_DIR}/Source/SurroundEditor.h"
```
Comment (McMartin) [inline examples/PlugInSamples/Surround/CMakeLists.txt]: @MartyLake: please indent with 2 spaces, thanks!

File: examples/PlugInSamples/NoiseGate/CMakeLists.txt (hunk 1 above)
Comment (MartyLake) [inline examples/PlugInSamples/NoiseGate/CMakeLists.txt]: 👍

File: examples/PlugInSamples/NoiseGate/CMakeLists.txt (hunk 2 above)
Comment (MartyLake) [inline examples/PlugInSamples/NoiseGate/CMakeLists.txt]: 👍

File: examples/PlugInSamples/Surround/CMakeLists.txt (hunk 3 above)
Comment (MartyLake) [inline examples/PlugInSamples/Surround/CMakeLists.txt]: :+1:

**Comment by MartyLake** [comment]
@McMartin just applied the modifications, please review thanks

**Comment by McMartin** [comment]
@MartyLake: please split the last commit into each previous commit and reword the commits to match `Support building JUCE's {xyz} example`.

**Comment by MartyLake** [comment]
@McMartin : done ! please review thanks

File: examples/PlugInSamples/GainPlugIn/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+  PLUGIN_MIDI_OUTPUT OFF
+  MIDI_EFFECT_PLUGIN ON
+  KEY_FOCUS ON
+  PLUGIN_AU_EXPORT_PREFIX "GainPlugInAU"
+  # PLUGIN_AU_MAIN_TYPE
+  # VST_CATEGORY
+)
+
+jucer_project_files("GainPlugIn/Source"
+    "${GainPlugIn_jucer_DIR}/Source/GainProcessor.cpp"
```
Comment (McMartin) [inline examples/PlugInSamples/GainPlugIn/CMakeLists.txt]: @MartyLake: please indent with 2 spaces, thanks!

File: examples/PlugInSamples/NoiseGate/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+)
+
+if(NOT EXISTS "${NoiseGate_jucer_FILE}")
+  message(FATAL_ERROR "No such file: ${NoiseGate_jucer_FILE}")
+endif()
+
+get_filename_component(NoiseGate_jucer_DIR "${NoiseGate_jucer_FILE}" DIRECTORY)
+
+
+
```
Comment (McMartin) [inline examples/PlugInSamples/NoiseGate/CMakeLists.txt]: @MartyLake: there should be 2 empty consecutive lines maximum.

**Comment by MartyLake** [comment]
done ! please review thanks :) @McMartin

File: examples/PlugInSamples/Arpeggiator/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,186 @@
+# Copyright (c) 2016 Matthieu Talbot
```
Comment (McMartin) [inline examples/PlugInSamples/Arpeggiator/CMakeLists.txt]: @MartyLake: please change to 2017, thanks! (same thing in the other files)

**Comment by McMartin** [comment]
@MartyLake: the commit message title doesn't have a period (`.` ) at the end normally.

**Pull request merged**